devito/cgen_utils.py

```diff
@@ -171,3 +172,6 @@ def ccode(expr, **settings):
 printvar = lambda i: c.Statement('printf("%s=%%s\\n", %s); fflush(stdout);' % (i, i))
 INT = Function('INT')
 FLOAT = Function('FLOAT')
+DOUBLE = Function('DOUBLE')
+
+cast_mapper = {np.float32: FLOAT, float: DOUBLE, np.float64: DOUBLE}
```

Comment:
Now that I see that, would it be a good time to drop the hard coded ` +F` for constants?

Reply:
yes, however I **think** that if you're doing ``float*double``, then the ``float`` gets automatically promoted to ``double``. But I should double check this.